### PR TITLE
bpo-30615: Fix the leak reference in Modules/_elementtree.c

### PR DESCRIPTION
**Pull request by matrixise**

### CODE REVIEW

**Review by vstinner — CHANGES_REQUESTED** [review]
This change it wrong: don't call Py_ReprLeave() is status > 0.

**Comment by matrixise** [comment]
that's the reason of my request for the review, I wanted to know if we need to call `Py_ReprLeave` when status > 0, or only if `status == -1`

**Comment by vstinner** [comment]
There is https://docs.python.org/dev/c-api/exceptions.html#c.Py_ReprEnter doc, it may help.

File: Modules/_elementtree.c [hunk 1]
```diff
@@ -1266,19 +1266,27 @@ element_repr(ElementObject* self)
...
-    if (status > 0)
+
+    if (status > 0) {
         PyErr_Format(PyExc_RuntimeError,
                      "reentrant call inside %s.__repr__",
                      Py_TYPE(self)->tp_name);
+    }
+
+    if (status == -1) {
+        Py_ReprLeave((PyObject *)self);
```
Comment (vstinner) [inline Modules/_elementtree.c]: Py_ReprLeave() must only be called if Py_ReprEnter() succeeded: status == 0.

**Review by vstinner — CHANGES_REQUESTED** [review]

**Review by vstinner — APPROVED** [review]

**Pull request merged**